hy/cmdline.py

```diff
@@ -73,8 +73,20 @@ def __init__(self, spy=False, output_fn=None, locals=None,
         code.InteractiveConsole.__init__(self, locals=locals,
                                          filename=filename)
 
+        # Pre-mangle symbols for repl recent results: *1, *2, *3
+        self._repl_results_symbols = [mangle("*{}".format(i + 1)) for i in range(3)]
+        self.locals.update({sym: None for sym in self._repl_results_symbols})
+
     def runsource(self, source, filename='<input>', symbol='single'):
         global SIMPLE_TRACEBACKS
+
+        def error_handler(e, use_simple_traceback=SIMPLE_TRACEBACKS):
```

Comment:
Be careful of how positional arguments work here. The default value is created once and shared between all function calls:

```python
>>> TEST = "Hello!"
>>> def foobar():
...     global TEST
...     
...     def inner(test=TEST):
...         print(test)
...         
...     inner()
...     TEST = "ANOTHER MESSAGE"
...     inner()
... 
>>> foobar()
Hello!
Hello!
```

Technically this should be fine because we only read from it, but its something linters will complain about.

I think it be best to just make it an explicit argument always, since you rely on the implicit default twice and explicitly set it twice.

Reply:
Thanks, I didn't know that could be a concern....

Just updated the code and make the default value to `False`. I am thinking that to use the key value when calling `error_handler`, so the code is easier to reason. made it default to false to void verbose.